DuckDuckGoTests/AutoClearLogicTests.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  AutoClearLogicTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2017 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+@testable import DuckDuckGo
+@testable import Core
+
+class AutoClearLogicTests: XCTestCase {
+    
+    class MockWorker: AutoClearWorker {
+        
+        var forgetDataExpectation: XCTestExpectation?
```

Comment:
I'm not a huge fan of this.  I understand the motivation, but I don't think it's particularly clear, especially if a test fails. 

For example, if you change this code to make the test fail and run it you get the test failed markers in Xcode but you don't get the inline reason for the error, you have to go hunting for it in the issue navigator.  Also it's adding a delay to the tests when they do fail.

Personally, I think asserts are clearer (for the reader) and the tests will fail faster. Can you change this to match how we've done mocks elsewhere, please?

This is another one to add to our future "architecture" discussion.  :)

Reply:
Replaced with simple invocation counter. :)